packages/itwinui-css/src/input/input.scss

```diff
@@ -3,51 +3,32 @@
 @use '../mixins';
 
 @mixin iui-input {
-  --_iui-input-background-color: var(--iui-color-background);
-  --_iui-input-border-color: var(--iui-color-border-foreground);
-  --_iui-input-text-color: var(--iui-color-text);
+  --_iui-field-color-text-placeholder: var(--iui-color-text-placeholder);
 
-  @include mixins.iui-reset;
-  @include mixins.iui-focus($offset: -2px, $thickness: 2px);
   inline-size: 100%;
-  font-family: inherit;
-  font-size: var(--iui-font-size-1);
-  font-weight: var(--iui-font-weight-normal);
   line-height: var(--iui-size-l);
-  border-radius: var(--iui-border-radius-1);
   appearance: none;
-  min-block-size: var(--_iui-input-min-height);
   padding-block: var(--_iui-input-padding-block);
   padding-inline: var(--iui-size-s);
-  color: var(--_iui-input-text-color);
-  background-color: var(--_iui-input-background-color);
-  border: 1px solid var(--_iui-input-border-color);
-  transition: border-color var(--iui-duration-1) ease-out;
   @include iui-input-size;
 
   &::placeholder {
-    --_iui-input-text-color: var(--iui-color-text-placeholder);
+    color: var(--_iui-field-color-text-placeholder);
     font-weight: var(--iui-font-weight-light);
     opacity: 1;
     user-select: none;
-    color: var(--_iui-input-text-color);
   }
 
-  &:autofill {
-    --_iui-input-background-color: var(--iui-color-background-accent-muted);
-    --_iui-input-border-color: var(--iui-color-border-accent);
-    --_iui-input-text-color: var(--iui-color-text-accent);
-  }
-
-  &:hover {
-    --_iui-input-border-color: var(--iui-color-border-foreground-hover);
+  &:where([disabled], :disabled, [aria-disabled='true']) {
+    &::placeholder {
+      --_iui-field-color-text-placeholder: transparent;
```

Comment:
I think will warrant a discussion.

Before:
- Disabled with value = text color
- Disabled with placeholder = placeholder text color

After:
- Disabled with value = disabled text color
- Disabled with placeholder = transparent, not shown

This was done because it differed from the disabled styling on buttons.  If we were to revert this change, we should maybe consider making changes to `.iui-field` to not handle disabled state or at least handle less.

Reply:
I'm not sure if this will lead to any accessibility issues. Because looks like the [color contrast criterion](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) applies also to placeholder text too?

<img width="815" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/a9a1580e-50db-4ccb-a015-e44a742dbf9f">

I also tried looking at the HTML spec for [placeholder](https://html.spec.whatwg.org/multipage/input.html#the-placeholder-attribute) and [disabled](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#enabling-and-disabling-form-controls:-the-disabled-attribute) but couldn't find any relevant information.

Since the native disabled `<input>`s show the placeholder ([playground](https://codepen.io/rohan-kadkol/pen/MWRGLaB?editors=1000)), I guess we should show it too?

/cc @iTwin/itwinui-1 @iTwin/itwinui-2

---

Also, if we are indeed going ahead with making the placeholder transparent on disabled inputs, I believe we would need a changeset for this.

Reply:
Why are we setting transparent? That's a terrible idea and I'm genuinely shocked to see it 😵‍💫

Lets keep the placeholder text color. CSS should not be used this way. The correct way to handle this would be using a combination of TS/React and documentation.

Reply:
Removed `transparent` styling from disabled placeholder text.  If we're going to remove the disabled placeholder text via TS/React, then I assume I should remove the CSS demos that show that?

Reply:
I'm fine with keeping the demos (they're only used for visual tests) or with removing them (since we don't want to support this scenario).

Reply:
Ok.  Styling has been reverted, CSS demos have been removed, and changeset has been updated.